modules/passkey/src/utils/webauthn.ts

```diff
@@ -0,0 +1,185 @@
+    attestationObject.authData.byteOffset,
+    attestationObject.authData.byteLength,
+  )
+  const credentialIdLength = authData.getUint16(53)
+  const cosePublicKey = attestationObject.authData.slice(55 + credentialIdLength)
+  const key: Map<number, unknown> = CBOR.decode(cosePublicKey)
+  const bn = (bytes: Uint8Array) => BigInt(ethers.hexlify(bytes))
+  return {
+    x: bn(key.get(-2) as Uint8Array),
+    y: bn(key.get(-3) as Uint8Array),
+  }
+}
+
+/**
+ * Decode the additional client data JSON fields. This is the fields other than `type` and
+ * `challenge` (including `origin` and any other additional client data fields that may be added by
+ * the authenticator).
+ *
+ * See <https://w3c.github.io/webauthn/#clientdatajson-serialization>
+ */
+export function decodeClientDataFields(response: Pick<AuthenticatorAssertionResponse, 'clientDataJSON'>): string {
+  const clientDataJSON = new TextDecoder('utf-8').decode(response.clientDataJSON)
+  const match = clientDataJSON.match(/^\{"type":"webauthn.get","challenge":"[A-Za-z0-9\-_]{43}",(.*)\}$/)
+
+  if (!match) {
+    throw new Error('challenge not found in client data JSON')
+  }
+
+  const [, fields] = match
+  return fields
+}
+
+/**
+ * Decode the signature R and S values from the authenticator response's DER-encoded signature.
+ *
+ * Note that this implementation assumes that all lengths in the DER encoding fit into 8-bit
+ * integers, which is true for the kinds of signatures we are decoding but generally false. I.e.
+ * **this code should not be used in any serious application**.
+ *
+ * See:
+ * - <https://datatracker.ietf.org/doc/html/rfc3279#section-2.2.3>
+ * - <https://en.wikipedia.org/wiki/X.690#BER_encoding>
+ */
+export function decodeSignature(response: Pick<AuthenticatorAssertionResponse, 'signature'>): { r: bigint; s: bigint } {
+  const view = new DataView(response.signature)
+
+  const check = (x: boolean) => {
+    if (!x) {
+      throw new Error('invalid signature encoding')
+    }
+  }
+  const readInt = (offset: number) => {
+    check(view.getUint8(offset) === 0x02)
+    const len = view.getUint8(offset + 1)
+    const start = offset + 2
+    const end = start + len
+    const n = BigInt(ethers.hexlify(new Uint8Array(view.buffer.slice(start, end))))
+    check(n < ethers.MaxUint256)
+    return [n, end] as const
+  }
+
+  // check that the sequence header is valid
+  check(view.getUint8(0) === 0x30)
+  check(view.getUint8(1) === view.byteLength - 2)
+
+  // read R and S
+  const [r, sOffset] = readInt(2)
+  const [s] = readInt(sOffset)
+
+  return { r, s }
+}
+
+/**
+ * Encodes the given WebAuthn signature into a string. Used for testing purposes.
+ *
+ * @param authenticatorData - The authenticator data as a Uint8Array.
+ * @param clientDataFields - The client data fields as a string.
+ * @param r - The value of r as a bigint.
+ * @param s - The value of s as a bigint.
+ * @returns The encoded string.
+ */
+export function getSignatureBytes({
```

Comment:
Unrelated to this PR, but from the documentation, shouldn't this method be called 
```suggestion
export function encodeSignatureBytes({
```

Don't have to change it here, just a small detail I noticed in passing.

Reply:
Will keep it unresolved to change in the next PR 👍🏾